### DISCUSSION TITLE
mix-trunks on ioll2

### DISCUSSION THREAD

**Discussion by DanPartelly** [ipspace/netlab#1456] [General]
Im trying to enable mixed-trunk on IOL2. Working with 62-vlan-mixed-trunk.yml

However the generated initial configuration cmakes E0/1 routed,  then creates a subinterface on E0/1 (the link between s1-s2) and assigns an IP on it, probably because vlan blue hes route mode  and then, in the vlan.conf tries to convert this interface in a dot1q trunk. Which is not legal, a routed port can't be a trunk. Later ospf.conf tries to run routing on said subinterface. 

Any recommendations on where to start to tackle this ? Im a bit out of my depth.

**Comment by ipspace** [comment]
The "mixed routed/bridged trunk" model works on routers that enable bridging between VLAN subinterfaces. For example, we configure subinterface on all trunk ports on Cisco IOS config and then add individual interfaces to bridge domains.

The layer-3 switches (Arista EOS, NX OS, IOSvL2) usually require an all-or-nothing "switchport" model (either a physical interface has routed subinterfaces or it's a switch port and has bridged VLANs) and thus cannot support the mixed trunk.

IOLL2 probably uses a layer-3 switch configuration model, so I wouldn't expect it to support mixed trunks.

**Reply by DanPartelly** [reply]
Thank you. Ill stop here then. 01- to 61-vlan-routed-native (included) of vlan tests are passed by ioll2 image. You'll be able to reproduce my results soon. Ive run all day an instrumented build of my iouyap fork with address and undefined behavior sanitizer enabled.  Ive had to fix a buffer underflow on stack. But now, the main loops run for hours with no reports. Ill proceed to a release phase. 
> 
> IOLL2 probably uses a layer-3 switch configuration model, so I wouldn't expect it to support mixed trunks.

**Comment by ipspace** [comment]
I think we got as far as we could
